metapool/calculate_coefficients.py

```diff
@@ -0,0 +1,147 @@
+import scipy
+import seaborn as sns
+import pandas as pd
+import numpy as np
+
+
+def plot_with_fit(xcol, ycol, groups, xmax=4, ymax=4,
+                  textpos=(.6, .2)):
+    from matplotlib import pyplot as plt
+    import matplotlib
+    matplotlib.use("TkAgg")
+    x = 0
+    y = 0
+    fig, axes = plt.subplots(xmax, ymax, sharex=True, sharey=True)
+    for name, group in groups:
+        ax = axes[y, x]
+        # ax.scatter(group["CPMlog"], group["Dilution2"])
+
+        # TODO FIXME HACK: Unclear what R^2 value is being printed by ggplot
+        #  but it definitely isn't a pearson computed per group.  *shrug*
+        no_nan = group.dropna()
+        r, p = scipy.stats.pearsonr(no_nan[xcol], no_nan[ycol])
+        sns.regplot(x=xcol, y=ycol, data=group, ax=ax)
+        ax.text(textpos[0], textpos[1], 'r={:.2f}\np={:.2g}'.format(r, p),
+                transform=ax.transAxes)
+        if x > 0:
+            ax.set_ylabel("")
+        if y < ymax - 1:
+            ax.set_xlabel("")
+
+        ax.set_title("\n".join(name))
+        x += 1
+        if x == xmax:
+            x = 0
+            y += 1
+        if y == ymax:
+            plt.show()
+            x = 0
+            y = 0
+
+    if x != 0 or y != 0:
+        plt.show()
```

Comment:
Just to confirm, this is showing each of the panels as it loops through `groups`, right? Then what line 41/42 do? Just to confirm, you don't need to `.show()` when `x == xmax`, right?

Reply:
41/42 allow you to set xmax and ymax independent of the number of plots being made without truncating prior to the plots that would be shown on the last page.  If x != 0 or y != 0, there are still plots that have not yet been shown.